src/Entity/Post.php

```diff
@@ -78,12 +70,11 @@ class Post
     #[ORM\ManyToMany(targetEntity: Tag::class, cascade: ['persist'])]
     #[ORM\JoinTable(name: 'symfony_demo_post_tag')]
     #[ORM\OrderBy(['name' => 'ASC'])]
-    #[Assert\Count(max: 4, maxMessage: 'post.too_many_tags')]
     private Collection $tags;
 
-    public function __construct()
+    public function __construct(User $author)
     {
-        $this->publishedAt = new \DateTimeImmutable();
```

Comment:
Why have you removed this? So, now the publish date is not set when the entity is constructed.

Reply:
This was initially set up to provide a default value during the Post creation form, but its final value was always set through the setter, because it's currently a user input.

Now the default value is set in the DTO instead, and the `Post::$publishedAt` value continues to be set through its `setPublishedAt()` setter method.

Note that (as before) this proposal does not follow the stricter path to guarantee that the entity’s state is always valid from its construction which I'd like to (waiting for confirmation)